crates/core/src/metrics.rs

```diff
@@ -0,0 +1,13 @@
+use crate::error::CarbonResult;
+use async_trait::async_trait;
+
+#[async_trait]
+pub trait Metrics: Send + Sync {
+    async fn initialize_metrics(&self) -> CarbonResult<()>;
```

Comment:
nit: this should be just `initialize`, `flush`, `shutdown`
when i was mentioning function renaming, i was talking about the functions in the `Pipeline` implementation